### PR TITLE
Disallow using portLevelMtls at namespace/mesh policy

### PR DESCRIPTION
**Pull request by diemtvu**
Issue: https://github.com/istio/istio/issues/20746

### CODE REVIEW

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -1734,6 +1734,16 @@ var ValidatePeerAuthentication = registerValidateFunc("ValidatePeerAuthenticatio
...
 		}
 
+		if emptySelector && len(in.PortLevelMtls) != 0 {
+			errs = appendErrors(errs,
+				fmt.Errorf("default peer authentication cannot have port level mTLS"))
+		}
+
+		if in.PortLevelMtls != nil && len(in.PortLevelMtls) == 0 {
+			errs = appendErrors(errs,
+				fmt.Errorf("port level mTLS, if define, must have at most one element"))
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go]: at least

Comment (diemtvu) [reply pkg/config/validation/validation.go]: Good catch. Thanks

File: pkg/config/validation/validation.go [hunk 2]
```diff
@@ -1734,6 +1734,16 @@ var ValidatePeerAuthentication = registerValidateFunc("ValidatePeerAuthenticatio
 				fmt.Errorf("peer authentication with empty workload selector must be named %q", constants.DefaultAuthenticationPolicyName))
 		}
 
+		if emptySelector && len(in.PortLevelMtls) != 0 {
+			errs = appendErrors(errs,
+				fmt.Errorf("default peer authentication cannot have port level mTLS"))
```
Comment (incfly) [inline pkg/config/validation/validation.go]: `default` here does it mean mesh/namespace level authn? can we reword to `mesh/namespace peer authentication cannot have...` to be more specific?

"default" is often overloaded i feel

File: pkg/config/validation/validation.go [hunk 3]
```diff
@@ -1734,6 +1734,16 @@ var ValidatePeerAuthentication = registerValidateFunc("ValidatePeerAuthenticatio
...
+		}
+
+		if in.PortLevelMtls != nil && len(in.PortLevelMtls) == 0 {
+			errs = appendErrors(errs,
+				fmt.Errorf("port level mTLS, if define, must have at least one element"))
```
Comment (incfly) [inline pkg/config/validation/validation.go]: if defined

File: pkg/config/validation/validation.go (hunk 2 above)
Comment (incfly) [reply pkg/config/validation/validation.go]: peer authn.. without selector cannot have ...

Comment (diemtvu) [reply pkg/config/validation/validation.go]: done

File: pkg/config/validation/validation.go (hunk 3 above)
Comment (diemtvu) [reply pkg/config/validation/validation.go]: done

**Pull request merged**